core/domain/acl_decorators.py

```diff
@@ -68,6 +76,39 @@ def check_activity_accessible(
             activity_rights.viewable_if_private)
 
 
+def check_exploration_editable(user_id, user_actions, exploration_rights):
+    """Returns a boolean to signify whether given exploration is editable
+    by the user or not.
+
+    Args:
+        user_id: str. Id of the given user.
+        user_actions: list(str). List of actions given user can perform.
+        exploration_rights: Exploration rights object.
+
+    Returns:
+        bool. Whether the given exploration can be accessed.
+    """
+    if exploration_rights.community_owned:
+        return True
+
+    if role_services.ACTION_EDIT_ANY_EXPLORATION in user_actions:
+        return True
+
+    if exploration_rights.status == rights_manager.ACTIVITY_STATUS_PUBLIC:
+        if (
```

Comment:
Indentation looks a bit weird. How about:

```
        if (role_services.ACTION_EDIT_ANY_PUBLIC_EXPLORATION in
                user_actions):
```

-- does that work? If not, then maybe declare:

```
    can_edit_any_public_exploration = ...
    if can_edit_any_public_exploration:
        ...
```

Ditto below (in multiple places).